tests/basic/bug68986.phpt

```diff
@@ -0,0 +1,23 @@
+--TEST--
+Bug #68986 (pointer returned by php_stream_fopen_temporary_file not validated in memory.c)
+--INI--
+default_charset=UTF-8
+--FILE--
+<?php
+if (substr(PHP_OS, 0, 3) == 'WIN') {
+	die('skip.. only for unix');
+}
+
+$tempDir = getenv("TMPDIR");
+mkdir($tempDir . '/php68986');
+system("chmod 444 " . $tempDir . '/php68986');
+putenv("TMPDIR=" . $tempDir . '/php68986');
```

Comment:
maybe there has been a miss-understanding of the permission in the test case 

the permission of a directory only applies to its children. when removing the directory the parent directories permission is applied. hence a read only directory can be removed if the parent directory has writable permission. 

@laruence @smalyshev i can post a gist of a improved test case if u like. probably using the `sys_get_temp_dir()` let me know if you still need a test case